pkg/hash/hash.go

```diff
@@ -3,100 +3,32 @@
 package hash
 
 import (
-	"crypto"
-	"fmt"
-	"strings"
-
-	"github.com/nix-community/go-nix/pkg/nixbase32"
-)
-
-//nolint:revive
-type HashType string
-
-const (
-	HashTypeSha256 = "sha256"
-	HashTypeSha512 = "sha512"
+	"hash"
 )
 
+// Hash can be used to calculate and display various hashes.
+// It implements the io.Writer interface, which will update
+// the internal hash state.
+// It can also be used to parse existing digests.
+// In this case, the writer interface is not available.
 type Hash struct {
-	HashType HashType
-	Digest   []byte
-}
-
-// hashFunc returns the cryptographic hash function for the passed HashType (implementing crypto.Hash)
-// It panics when encountering an invalid HashType, as these can only occur by
-// manually filling the struct.
-func hashFunc(hashType HashType) crypto.Hash {
-	switch hashType {
-	case HashTypeSha256:
-		return crypto.SHA256
-	case HashTypeSha512:
-		return crypto.SHA512
-	default:
-		panic(fmt.Sprintf("Invalid hash type: %v", hashType))
-	}
-}
-
-// ParseNixBase32 returns a new Hash struct, by parsing a hashtype:nixbase32 string, or an error.
-func ParseNixBase32(s string) (*Hash, error) {
-	i := strings.IndexByte(s, ':')
-	if i <= 0 {
-		return nil, fmt.Errorf("unable to find separator in %v", s)
-	}
-
-	hashTypeStr := s[:i]
+	HashType int
 
-	var hashType HashType
-
-	switch hashTypeStr {
-	case HashTypeSha256:
-		hashType = HashTypeSha256
-	case HashTypeSha512:
-		hashType = HashTypeSha512
-	default:
-		return nil, fmt.Errorf("unknown hash type: %v", hashType)
-	}
-
-	// The digest afterwards is nixbase32-encoded.
-	// Calculate the length of that string, in nixbase32 encoding
-	digestLenBytes := hashFunc(hashType).Size()
-	encodedDigestLen := nixbase32.EncodedLen(digestLenBytes)
-
-	encodedDigestStr := s[i+1:]
-	if len(encodedDigestStr) != encodedDigestLen {
-		return nil, fmt.Errorf("invalid length for encoded digest line %v", s)
-	}
-
-	digest, err := nixbase32.DecodeString(encodedDigestStr)
-	if err != nil {
-		return nil, err
-	}
-
-	return &Hash{
-		HashType: hashType,
-		Digest:   digest,
-	}, nil
-}
-
-// MustParseNixBase32 returns a new Hash struct, by parsing a hashtype:nixbase32 string, or panics on error.
-func MustParseNixBase32(s string) *Hash {
-	h, err := ParseNixBase32(s)
-	if err != nil {
-		panic(err)
-	}
+	// Hash only populated if we construct the hash on our own.
+	hash hash.Hash
```

Comment:
Should this be called `hasher` instead?

Reply:
The interface is called `Hash` so I think it's fine.